config/main.py

```diff
@@ -1313,5 +1315,274 @@ def naming_mode_alias():
+
+    if not sflow_tbl:
+        sflow_tbl = {'global': {'admin_state': 'down'}}
+    else:
+        sflow_tbl['global']['admin_state'] = 'down'
+
+    config_db.mod_entry('SFLOW', 'global', sflow_tbl['global'])
+
+#
+# 'sflow' command ('config sflow polling-interval ...')
+#
+@sflow.command('polling-interval')
+@click.argument('interval',  metavar='<polling_interval>', required=True,
+                type=int)
+@click.pass_context
+def polling_int(ctx, interval):
+    """Set polling-interval for counter-sampling (0 to disable)"""
+    if interval not in range(5, 301) and interval != 0:
+        click.echo("Polling interval must be between 5-300 (0 to disable)")
+
+    config_db = ctx.obj['db']
+    sflow_tbl = config_db.get_table('SFLOW')
+
+    if not sflow_tbl:
+        sflow_tbl = {'global': {'admin_state': 'down'}}
+
+    sflow_tbl['global']['polling_interval'] = interval
+    config_db.mod_entry('SFLOW', 'global', sflow_tbl['global'])
+
+def is_valid_sample_rate(rate):
+    return rate in range(256, 8388608 + 1)
+
+
+#
+# 'sflow interface' group
+#
+@sflow.group()
+@click.pass_context
+def interface(ctx):
+    """Configure sFlow settings for an interface"""
+    pass
+
+#
+# 'sflow' command ('config sflow interface enable  ...')
+#
+@interface.command()
+@click.argument('ifname', metavar='<interface_name>', required=True, type=str)
+@click.pass_context
+def enable(ctx, ifname):
+    if not interface_name_is_valid(ifname) and ifname != 'all':
+        click.echo("Invalid interface name")
+        return
+
+    config_db = ctx.obj['db']
+    intf_dict = config_db.get_table('SFLOW_SESSION')
+
+    if intf_dict and ifname in intf_dict.keys():
+        intf_dict[ifname]['admin_state'] = 'up'
+        config_db.mod_entry('SFLOW_SESSION', ifname, intf_dict[ifname])
+    else:
+        config_db.mod_entry('SFLOW_SESSION', ifname, {'admin_state': 'up'})
+
+#
+# 'sflow' command ('config sflow interface disable  ...')
+#
+@interface.command()
+@click.argument('ifname', metavar='<interface_name>', required=True, type=str)
+@click.pass_context
+def disable(ctx, ifname):
+    if not interface_name_is_valid(ifname) and ifname != 'all':
+        click.echo("Invalid interface name")
+        return
+
+    config_db = ctx.obj['db']
+    intf_dict = config_db.get_table('SFLOW_SESSION')
+
+    if intf_dict and ifname in intf_dict.keys():
+        intf_dict[ifname]['admin_state'] = 'down'
+        config_db.mod_entry('SFLOW_SESSION', ifname, intf_dict[ifname])
+    else:
+        config_db.mod_entry('SFLOW_SESSION', ifname,
+                            {'admin_state': 'down'})
+
+#
+# 'sflow' command ('config sflow interface sample-rate  ...')
+#
+@interface.command('sample-rate')
+@click.argument('ifname', metavar='<interface_name>', required=True, type=str)
+@click.argument('rate', metavar='<sample_rate>', required=True, type=int)
+@click.pass_context
+def sample_rate(ctx, ifname, rate):
+    if not interface_name_is_valid(ifname) and ifname != 'all':
```

Comment:
No need to include key `all` to be valid here as sflowmgrd only respond to field `admin_state` under `SFLOW_SESSION|all`

Reply:
This check is to ensure the user gives a valid interface name or `all` before writing to the dB.